prog.go

```diff
@@ -783,7 +782,7 @@ func LoadPinnedProgram(fileName string, opts *LoadPinOptions) (*Program, error)
 		return nil, fmt.Errorf("info for %s: %w", fileName, err)
 	}
 
-	return &Program{"", fd, filepath.Base(fileName), fileName, info.Type}, nil
+	return &Program{"", fd, info.Name, fileName, info.Type}, nil
```

Comment:
I think this would leave > 4.11 && < 4.15 without a name. Maybe keep filepath.Base if `!haveProgName()`?

Reply:
Good point. I changed my patch to check for the necessary feature and fall back to the file base name, just as before.